tests/tests_lite/accelerators/test_cuda.py

```diff
@@ -61,3 +61,4 @@ def test_num_cuda_devices_without_nvml(*_):
     num_cuda_devices.cache_clear()
     assert is_cuda_available()
     assert num_cuda_devices() == 100
+    num_cuda_devices.cache_clear()
```

Comment:
This was causing

`E       RuntimeError: CUDA error: invalid device ordinal` with the command:

`pytest tests/tests_lite/accelerators/test_cuda.py tests/tests_lite/plugins/precision/test_native_amp_integration.py`